pkg/controller/atlasdatabaseuser/databaseuser.go

```diff
@@ -149,28 +151,27 @@ func performUpdateInAtlas(ctx *workflow.Context, k8sClient client.Client, projec
 func validateScopes(ctx *workflow.Context, projectID string, user mdbv1.AtlasDatabaseUser) error {
 	for _, s := range user.GetScopes(mdbv1.ClusterScopeType) {
 		var apiError *mongodbatlas.ErrorResponse
-		_, _, err := ctx.Client.Clusters.Get(context.Background(), projectID, s)
-		if errors.As(err, &apiError) && apiError.ErrorCode == atlas.ClusterNotFound {
+		_, _, regularErr := ctx.Client.Clusters.Get(context.Background(), projectID, s)
+		_, _, advancedErr := ctx.Client.AdvancedClusters.Get(context.Background(), projectID, s)
```

Comment:
the cluster could be either regular or advanced, we need to check for both

Reply:
If we check errors thought both API version shouldn't we return an error if either of them is an error, not both?
Also I thought that we can get a regular cluster through the advanced API, I can check this.